packages/Modal/stories/Modal.stories.js

```diff
@@ -72,6 +72,23 @@ storiesOf("Modal", module)
       <DemoFullWidthContent />
     </ModalStory>
   ))
+  .add("with form render", () => (
+    <Modal
+      isOpen
+      as="form"
+      onSubmit={event => {
+        alert("submit");
+        event.preventDefault();
+      }}
+    >
+      <Modal.Content>
+        <input type="text" deafultValue="your text" />
+      </Modal.Content>
+      <Modal.Footer>
+        <Button type="submit">Submit</Button>
```

Comment:
I think `isSubmit` prop is used for this. (clicking the button currently doesn't submit the form)

Reply:
abstracted logic for the input type submits into the Button to make it work. https://codesandbox.io/s/quiet-water-8khug